salt/modules/dockerng.py

```diff
@@ -1634,6 +1637,26 @@ def _valid_pid_mode():  # pylint: disable=unused-variable
                 'pid_mode can only be \'host\', if set'
             )
 
+    def _valid_labels():  # pylint: disable=unused-variable
+        '''
+        Must be a dict or a list of strings
+        '''
+        if kwargs.get('labels') is None:
+            return
+        try:
+            _valid_stringlist('labels')
+        except SaltInvocationError:
+            try:
+                _valid_dict('labels')
+            except SaltInvocationError:
+                raise SaltInvocationError(
+                    'labels can only be a list of strings or a dict containing'
+                    ' strings')
```

Comment:
Can there be more context in the error message, e.g. a `dockerng:` prefix and/or the name of the state where it's used?  Or is that handled by Salt automatically already?